ecpy_hqc_legacy/instruments/drivers/driver_tools.py

```diff
@@ -147,6 +146,80 @@ def wrapper(self, *args, **kwargs):
     return decorator
 
 
+class InstrJob(object):
+    """Object returned by instrument starting a long running job.
+
+    Parameters
+    ----------
+    condition_callable : Callable
+        Callable taking no argument and indicating if the job is complete.
+
+    expected_waiting_time : float
+        Expected waiting time for the task to complete in seconds.
+
+    cancel : Callable, optional
+        Function to cancel the task.
+
+    """
+    def __init__(self, condition_callable, expected_waiting_time, cancel):
+        self.condition_callable = condition_callable
+        self.expected_waiting_time = expected_waiting_time
+        self.cancel = cancel
+        self._start_time = time.time()
+
+    def wait_for_completion(self, break_condition_callable=None, timeout=15,
+                            refresh_time=1):
```

Comment:
to see if refreshing every second is not a bit much since communication with cs4 is sometimes slow.